File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        stack=[]
        answer=""
        for i in range(len(s)):
            if s[i]=="(":
                stack.append(i)
            else:
                if len(stack)==1:
                    answer+=s[stack.pop()+1:i]
                else:
                    stack.pop()     
  
        return answer
                

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        if (s == null || s.length() <= 2) {
            return "";
        }
        
        StringBuilder result = new StringBuilder();
        int count = 1;
        
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == '(') {
                count++;
                if (count >= 2) {
                    result.append(s.charAt(i));
                }
            } else {
                count--;
                if (count >= 1) {  
                    result.append(s.charAt(i));
                }
            }
        }
        
        return result.toString();
    }
}